### PR TITLE
Support rt record functions, select and zext in parser

### PR DESCRIPTION
**Pull request by RolandMacDoland**
This PR addresses the support for tagged or untagged runtime functions to record results from the device. Specifically:

- `int_record`
- `bool_record`
- `result_record`

Besides, it also allows for the support of `select` and `zext` gates.

### CODE REVIEW

File: tests/qir_test.py [hunk 1]
```diff
@@ -118,6 +117,126 @@ def test_wasm_only(self) -> None:
...
+                com.op.type == OpType.Conditional
+            ):  # Raising an error if I use .commands_of_type(OpType.Conditional)
+                conditionals.append(com)
+
+        conditional0 = conditionals[0]
+        assert conditional0.op.type == OpType.Conditional
+        assert conditional0.bits == list(output_register)
+        assert conditional0.op.value == 1
+        assert (
+            sum([int(n) * 2**k for k, n in enumerate(conditional0.op.op.values)])
```
Comment (cqc-alec) [inline tests/qir_test.py]: Why is the cast to `int` necessary?

File: tests/qir_test.py [hunk 2]
```diff
@@ -118,6 +117,126 @@ def test_wasm_only(self) -> None:
...
+            == 99
+        )
+
+        conditional1 = conditionals[1]
+        assert conditional1.op.type == OpType.Conditional
+        assert conditional1.bits == list(output_register)
+        assert conditional1.op.value == 0
+        assert (
+            sum([int(n) * 2**k for k, n in enumerate(conditional1.op.op.values)])
```
Comment (cqc-alec) [inline tests/qir_test.py]: ... and here?

File: pytket_qir/gatesets/base.py [hunk 3]
```diff
@@ -38,7 +38,6 @@ class OpName(Enum):
     AND = "and"
     OR = "or"
     XOR = "xor"
-    WASM = "wasm"
```
Comment (cqc-alec) [inline pytket_qir/gatesets/base.py:41]: Is this not needed anymore?

File: tests/qir_test.py (hunk 1 above)
Comment (RolandMacDoland) [reply tests/qir_test.py]: Reminiscent of the old debugging times. Removed.

File: tests/qir_test.py (hunk 2 above)
Comment (RolandMacDoland) [reply tests/qir_test.py]: Removed.

File: pytket_qir/gatesets/base.py (hunk 3 above)
Comment (RolandMacDoland) [reply pytket_qir/gatesets/base.py:41]: No. This name is now dynamically created [here](https://github.com/CQCL/pytket-qir/blob/90b44b57a05a0b3f585b901148f2b9fed13c80e0/pytket_qir/generator.py#L286).

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**